subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/TaskExecution.java

```diff
@@ -334,17 +334,14 @@ public void visitRegularInputs(InputVisitor visitor) {
     public void visitOutputs(File workspace, OutputVisitor visitor) {
         TaskProperties taskProperties = context.getTaskProperties();
         for (OutputFilePropertySpec property : taskProperties.getOutputFileProperties()) {
-            File outputFile = property.getOutputFile();
-            if (outputFile != null) {
```

Comment:
Here we had a null check before, but OutputFileValueSupplier.fromSupplier() returns a supplier that has `getValue()` marked as `@Nonnull`. Is that ok?

Reply:
Yeah, I also removed the `@Nullable` from `OutputFilePropertySpec.getOutputFile()`. I followed all the call paths, and we don't have that case anymore. `null` values are filtered out here:

https://github.com/gradle/gradle/blob/d0eee56936a044eafd5d7d45ace2b9a3eb4dd6af/subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/OutputUnpacker.java#L118-L120